WordPress/Classes/ViewRelated/NUX/SignupGoogleViewController.swift

```diff
@@ -25,80 +25,129 @@ class SignupGoogleViewController: LoginViewController {
+    }
+
+    private func displayGoogleSingleSignOnIfNeeded() {
+        guard !hasShownGoogle else {
+            return
         }
+
+        displayGoogleSingleSignOn()
+        hasShownGoogle = true
     }
 
-    private func showGoogleScreen() {
+    private func displayGoogleSingleSignOn() {
         GIDSignIn.sharedInstance().disconnect()
 
         // Flag this as a social sign in.
-        loginFields.meta.socialService = SocialServiceName.google
+        loginFields.meta.socialService = .google
 
         // Configure all the things and sign in.
-        GIDSignIn.sharedInstance().delegate = self
-        GIDSignIn.sharedInstance().uiDelegate = self
-        GIDSignIn.sharedInstance().clientID = WordPressAuthenticator.shared.configuration.googleLoginClientId
-        GIDSignIn.sharedInstance().serverClientID = WordPressAuthenticator.shared.configuration.googleLoginServerClientId
+        guard let googleSSO = GIDSignIn.sharedInstance() else {
+            DDLogError("Something is very, very, very off. Well done, Google.")
+            return
+        }
 
-        GIDSignIn.sharedInstance().signIn()
+        googleSSO.delegate = self
+        googleSSO.uiDelegate = self
+        googleSSO.clientID = WordPressAuthenticator.shared.configuration.googleLoginClientId
+        googleSSO.serverClientID = WordPressAuthenticator.shared.configuration.googleLoginServerClientId
+
+        googleSSO.signIn()
     }
 }
 
+
 // MARK: - GIDSignInDelegate
 
 extension SignupGoogleViewController: GIDSignInDelegate {
+
     func sign(_ signIn: GIDSignIn?, didSignInFor user: GIDGoogleUser?, withError error: Error?) {
         GIDSignIn.sharedInstance().disconnect()
 
-        guard let user = user,
-            let token = user.authentication.idToken,
-            let email = user.profile.email else {
-                WordPressAuthenticator.track(.signupSocialButtonFailure, error: error)
-                self.navigationController?.popViewController(animated: true)
-                return
+        guard let googleUser = user, let googleToken = googleUser.authentication.idToken, let googleEmail = googleUser.profile.email else {
+            WordPressAuthenticator.track(.signupSocialButtonFailure, error: error)
+            self.navigationController?.popViewController(animated: true)
+            return
         }
 
-        // Store the email address and token.
-        loginFields.emailAddress = email
-        loginFields.username = email
-        loginFields.meta.socialServiceIDToken = token
-        loginFields.meta.googleUser = user
+        updateLoginFields(googleUser: googleUser, googleToken: googleToken, googleEmail: googleEmail)
+        createWordPressComUser(googleUser: googleUser, googleToken: googleToken, googleEmail: googleEmail)
+    }
+}
+
 
+// MARK: - WordPress.com Account Creation Methods
+//
+private extension SignupGoogleViewController {
+
+    /// TODO: Not cool with this. Let's refactor LoginFields, when time permits.
```

Comment:
Let's leave all these changes for a different PR?

Reply:
Yes sir! that's the idea. Let's, instead, file an issue to do so, which is more ... tracking friendly?.
(And i'll nuke this comment).

Thank you!